commons-datastore/commons-datastore-mongodb/src/main/java/org/opencb/commons/datastore/mongodb/MongoDBCollection.java

```diff
@@ -226,21 +224,11 @@ private DataResult endWrite(long numMatches, long numInserted, long numUpdated,
     private <T> DataResult<T> privateFind(ClientSession clientSession, Bson query, Bson projection, Class<T> clazz,
                                            ComplexTypeConverter<T, Document> converter, QueryOptions options) {
         long start = startQuery();
-
-        Future<Long> countFuture = null;
-        if (options.getBoolean(QueryOptions.COUNT)) {
-            ExecutorService executor = Executors.newSingleThreadExecutor();
-            countFuture = executor.submit(() -> mongoDBNativeQuery.count(clientSession, query));
-        }
-
-        /**
-         * Getting the cursor and setting the batchSize from options. Default value set to 20.
-         */
-        FindIterable<Document> findIterable = mongoDBNativeQuery.find(clientSession, query, projection, options);
+        MongoDBIterator<Document> findIterable = mongoDBNativeQuery.find(clientSession, query, projection, options);
         MongoCursor<Document> cursor = findIterable.iterator();
```

Comment:
I don't think it is necessary doing findIterable.iterator(); The idea of having the MongoDBIterator is also to be able to iterate over the elements for you. I would remove the line 228, and would use findIterable.hasNext(); findIterable.next(); and findIterable.close();